044-cluster-wide-volume-usage-quota-management.md

```diff
@@ -0,0 +1,253 @@
+    - [Volume](#volume)
+    - [Volume Source](#volume-source)
+    - [Quota Source](#quota-source)
+    - [Throttle Factor](#throttle-factor)
+    - [Cluster Volume Source](#cluster-volume-source)
+      - [Admin Client Configuration](#admin-client-configuration)
+      - [Limit Type Configuration](#limit-type-configuration)
+    - [Fallback Throttle Factor](#fallback-throttle-factor)
+- [Metrics](#metrics)
+- [Rejected Alternatives](#rejected-alternatives)
+- [Affected Projects](#affected-projects)
+- [Compatibility](#compatibility)
+
+## Current situation
+
+The [kafka-static-quota-plugin](https://github.com/strimzi/kafka-quotas-plugin) applies byte-rate limits on client connections to
+individual brokers (thresholds derived from storage limits), slowing producers down when consumed storage is over the soft limit (but below the hard limit) and
+effectively pausing publication when reaching or exceeding the hard limit. This largely prevents out of disk scenarios when topics
+are replicated to all brokers thus there is largely similar disk usage between all brokers. Assuming all the brokers
+have similar disk usage levels, they will all apply rate limits at similar times and levels, effectively giving cluster-wide out-of-storage protection.
+
+This however provides limited protection to clusters with un-even distribution of topics and thus storage usage.
+Additionally, it ties the plug-in directly to storage usage, there are other factors which users may wish to respond to.
+
+As clusters scale up the likelihood of even topic distribution drops. When topics are not evenly distributed it
+is possible for replication from broker `1` to broker `2` will cause broker `2` to consume all available storage
+without triggering throttling of clients, as broker `1` disk usage remains acceptable.
+
+Addressing the effects of uneven topic distribution sounds like it should come under [KIP-73](https://cwiki.apache.org/confluence/display/KAFKA/KIP-73+Replication+Quotas). Unfortunately replication quotas
+are designed to manage the additional network load of migrating replicas between brokers, which does not address client publication
+leading to out of disk conditions through replication.
+
+Currently, the kafka-quotas-plugin considers the total quantity of storage and how much of that is consumed (
+see [issue#2](https://github.com/strimzi/kafka-quotas-plugin/issues/2)) when considering whether to apply throttling to
+clients. This is problematic with respect to handling disk failure for Just a Bunch Of Disks (JBOD) deployments [(KIP-112)](https://cwiki.apache.org/confluence/display/KAFKA/KIP-112%3A+Handle+disk+failure+for+JBOD) as the
+broker will take partitions offline when the volume they are stored on runs out of space. Which in the case of unbalanced usage
+between volumes can lead to a volume running out of storage without throttling being applied. The broker will go offline
+if all volumes are unavailable.
```

Comment:
We removed the sentence about the broker going offline as this was more like context rather than the problem.

Is the sentence about `can lead to a volume running out of storage without throttling being applied` enough? Or should we re-iterate why running out of storage is a problem?